### PR TITLE
fix(document): fix conversion from blob to file with iobuffer

### LINKED ISSUE #4008: Cannot dump imageblob to bytesio
**ENV**

Python 3.9
jina 2.5.0

**Describe the bug**

If i try to dump an image blob to a io.bytesio object an error is thrown

```python
from jina import Document
import io
d =  Document(uri='steam_data/image_store/8c/5b/8c5b265b9c533636.png')
output = io.BytesIO() 

(
    d
    .load_uri_to_image_blob()
    .dump_image_blob_to_file(output)
)
```

```pytb
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_12013/123594708.py in <module>
      5 
      6 (
----> 7     d
      8     .load_uri_to_image_blob()
      9     .dump_image_blob_to_file(output)

~/miniconda3/envs/steam3.9/lib/python3.9/site-packages/jina/types/document/mixins/image.py in dump_image_blob_to_file(self, file, channel_axis)
    100             blob = _move_channel_axis(self.blob, channel_axis, -1)
    101             buffer = _to_png_buffer(blob)
--> 102             fp.write(buffer)
    103         return self
    104 

AttributeError: 'nullcontext' object has no attribute 'write'
```

**Describe how you solve it**

Here 
https://github.com/jina-ai/jina/blob/99b1e1dba1562659be679735912564150c08fe3b/jina/types/document/mixins/image.py#L98

the returned object of the context is not used in this line
```python
 with fp:
```
if we change the code to
```python
with fp as fp_cxt:
    blob = _move_channel_axis(self.blob, channel_axis, -1)
    buffer = _to_png_buffer(blob)
    fp_cxt.write(buffer)
```

it seems to work for my case.
Im not sure of the broader consequences though, as i dont know what the nullcontext is used for here

### PR DESCRIPTION
**Pull request by Stubatiger**
Using now the context return object instead of the actual manager which fixes #4008

### CODE REVIEW

File: tests/unit/types/document/test_converters.py [hunk 1]
```diff
@@ -83,6 +84,18 @@ def test_buffer_to_blob():
     assert doc.blob.shape == (85, 152, 3)  # h,w,c
 
 
+def test_image_blob_to_file_with_buffer():
```
Comment (JoanFM) [inline tests/unit/types/document/test_converters.py:87]: please use `tmpdir` fixture if you plan to dump somewhere? where is this `file` found?

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by JoanFM** [comment]
Hey @Stubatiger thank you very much for the contribution, I just have a little comment about the test written? Could you check?

File: tests/unit/types/document/test_converters.py (hunk 1 above)
Comment (Stubatiger) [reply tests/unit/types/document/test_converters.py:87]: Hey im not sure what you mean, im not dumping anything to a file?
Im just using an inmemory bytesio object.
If you mean the "test.png" image path, i copie that from the test 'test_buffer_to_blob' above

Comment (JoanFM) [reply tests/unit/types/document/test_converters.py:87]: fair enough

**Pull request closed without merging**